molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java

```diff
@@ -234,7 +237,24 @@ public Repository add(EntityMetaData emd, RepositoryDecoratorFactory decoratorFa
 	@Override
 	public synchronized Repository addEntityMeta(EntityMetaData emd)
 	{
-		return add(emd, new NonDecoratingRepositoryDecoratorFactory());
+		if (emd.getDecorator() == null)
+		{
+			return add(emd, new NonDecoratingRepositoryDecoratorFactory());
+		}
+		else
+		{
+			return add(emd, new RepositoryDecoratorFactory()
+			{
+				@Override
+				public Repository createDecoratedRepository(Repository repository)
```

Comment:
Can you add unit test for addEntityMeta for the case when no or one decorator is available?